plugins/source/aws/resources/services/iam/access_details_fetch.go

```diff
@@ -0,0 +1,80 @@
+package iam
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/iam/types"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client/services"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/resources/services/iam/models"
+)
+
+func fetchIamAccessDetails(ctx context.Context, res chan<- interface{}, svc services.IamClient, arn string) error {
+	config := iam.GenerateServiceLastAccessedDetailsInput{
+		Arn:         &arn,
+		Granularity: types.AccessAdvisorUsageGranularityTypeActionLevel,
+	}
+	output, err := svc.GenerateServiceLastAccessedDetails(ctx, &config)
+	if err != nil {
+		return err
+	}
+
+	getDetails := iam.GetServiceLastAccessedDetailsInput{
+		JobId: output.JobId,
+	}
+	for {
+		details, err := svc.GetServiceLastAccessedDetails(ctx, &getDetails)
+		if err != nil {
+			return err
+		}
+
+		requestTime := 0
+		switch details.JobStatus {
+		case types.JobStatusTypeInProgress:
+			time.Sleep(time.Millisecond * 200)
```

Comment:
How long do these reports take to generate on average? Seconds, minutes, hours? Just wondering if 200ms might be too frequent, maybe we should do every second or so.